### ISSUE TITLE
Compile error with using Specio types and roles being consumed multiple times

### ISSUE THREAD

**Issue by mgruner** [houseabsolute/Specio#18]
Please consider this following example that works correctly:

```perl
package test::RoleA;

use Specio::Declare;
use Specio::Library::Builtins;
use Moose::Role;

has 'A' => (
    is => 'rw',
    isa => 'HashRef',
);

package test::RoleB;
use Moose::Role;
with 'test::RoleA';

package test;
use Moose;
with 'test::RoleA', 'test::RoleB';

test->new()->A({});
```

This works well. If I use now `t('HashRef')` instead of just `'HashRef'`, I get a compile error.

```perl 
package test::RoleA;

use Specio::Declare;
use Specio::Library::Builtins;
use Moose::Role;

has 'A' => (
    is => 'rw',
    isa => t('HashRef'),
);

package test::RoleB;
use Moose::Role;
with 'test::RoleA';

package test;
use Moose;
with 'test::RoleA', 'test::RoleB';

test->new()->A({});
``` 

```txt
shell> perl test.pl
Operation "eq": no method found,
        left argument in overloaded package Specio::Constraint::Parameterizable,
        right argument in overloaded package Specio::Constraint::Parameterizable at /usr/local/Cellar/perl/5.32.1/lib/perl5/site_perl/5.32.1/darwin-thread-multi-2level/Moose/Meta/Role/Attribute.pm line 123.
```

This only occurs if `test::RoleA` is consumed twice, once directly in package `test` (which is unneeded in this case, but should work correctly), and once as a dependency of `test::RoleB`.

I hope this is the right place to report this issue. Thank you very much for considering.

**Comment by autarch** [comment]
Wow, that's a weird bug. I will have to find some time to deep dive into the Specio and Moose code to fix this, I suspect.

**Comment by autarch** [comment]
This is fixed in v0.47. Thanks for the bug report!

**Comment by mgruner** [comment]
Thanks a lot for the fast response @autarch! 🚀
